### ISSUE TITLE
A wayfinding bug in the map whose building is separate. 

### ISSUE THREAD

**Issue by ghost** [JulienNigon/SpellcasterUniversity-Bug-Report#1226]
The stupid student attempts to reach another building directly 
![bug](https://user-images.githubusercontent.com/64409561/116161308-1d790b80-a726-11eb-8db9-ab2520ecc9dc.png)
on the high layer, which leads to be stuck in the wall.

**Comment by Provideniya** [comment]
Yeah, noticed this myself last night (3/7/21) while playing on Volcano map. Unfortunately I didn't take any pictures.
Like with OP, this happened to me on higher floors - Floors 10 and 11, I think.

Everything was good until I placed rooms with left/right access (Janitor and Teachers Apartment) on the same level as the Dormitory and Students Hall on the other side of that.. I think.. 2 block wide gap. After placing those, both rooms started to fill with Students trying to get their Sleep (zZz icon) and Boredom (Kite icon) taken care of.

It was clear they were trying to travel from Janitor/Apartment to Dormitory/Students Hall, some even tried to use the doorway. They clearly thought those rooms were connected. Before (with 1 block larger gab between Dormi/SH and rest of the School, before placing the Janitor and Apartment) they gladly took the long way around or outright left the school grounds.

I noticed how almost all characters getting stuck first tried the side door and started to slide slightly towards the back of the room, like heading towards the rooms on the other side that are slightly further back than the Main Building - getting stuck into the wall in progress.
